corehq/tests/noseplugins/redislocks.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from datetime import datetime
+from threading import Lock
+from unittest.mock import patch
+
+import attr
+from nose.plugins import Plugin
+
+import dimagi.utils.couch as module
+from dimagi.utils.couch import get_redis_client
+
+from corehq.tests.noseplugins.uniformresult import uniform_description
+
+log = logging.getLogger(__name__)
+_LOCK = Lock()
+
+
+class RedisLockTimeoutPlugin(Plugin):
+    """A plugin that causes blocking redis locks to error on lock timeout"""
+    name = "test-redis-locks"
+    enabled = True
+
+    def configure(self, options, conf):
+        """Do not call super (always enabled)"""
+        self.patch = patch.object(module, "get_redis_client", TestRedisClient)
+
+    def startTest(self, case):
+        assert _LOCK.acquire(blocking=False), \
+            f"{uniform_description(case.test)}: concurrent tests not supported"
+        self.patch.start()
+
+    def stopTest(self, case):
+        self.patch.stop()
+        _LOCK.release()
+
+
+class TestRedisClient:
```

Comment:
what's the difference between this class and `TestRedisClient` in `locks.py`? assume it's by design that they have the same name? If not, may be worth renaming one of them for clarity

Reply:
Thanks for pointing that out. I can likely unify them so they use the same class. I'll do that in a follow-up PR.

Reply:
https://github.com/dimagi/commcare-hq/pull/27402/commits/2f5497487a13066bbd95f0aa091c0c240986777d